src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/GoToDefinition/RazorComponentDefinitionHelpers.cs

```diff
@@ -169,6 +168,7 @@ static bool TryGetTagName(RazorSyntaxNode node, [NotNullWhen(true)] out RazorSyn
                 return null;
             }
 
+            var csharpText = codeDocument.GetCSharpSourceText();
```

Comment:
It feels a bit fishy to assume that the `RazorCodeDocument` has exactly the same text that `csharpSyntaxTree` is based on.

Reply:
We could get the text from the syntax tree, but we still need the `CSharpDocument` below, so can't avoid the code document altogether. Could pass in the `RazorCSharpDocument` instead of the `RazorCodeDocument`, but either way there still needs to be an assumption that they originate from the same place.

Personally speaking I don't see this as a big problem though. I can doc the method to make the expectations clear, but if anyone does something weird  and sends in mis-matching data, they probably won't get a good result and their tests will fail etc. ie, yes its a footgun, but it can only affect our feet.

Reply:
Sure. I just feel a little oogy when the expectations are murky. It's probably fine for internal code, but it wouldn't make it through a .NET API review.

Reply:
Originally I was against having a `GetCSharpSyntaxTree` on `IDocumentSnapshot`, because it seemed like introducing a similar "grab bag of helpers" like we have in `DocumentContext`, but maybe I'm coming around to it. eg, Moving `TryGetPropertyRangeAsync` back to the component definition service would solve this issue, because it could be passed just the document snapshot, and then it can get the syntax tree from there... but it could also do that if there was just a way to get a syntax tree from a document snapshot.